File: vmmjava/vmm/VmmException.java
package vmm;

/**
 * Runtime Exception used to indicate exceptions from the underlying native API.
 * @see https://github.com/ufrisk/MemProcFS
 * @author Ulf Frisk - pcileech@frizk.net
 */
public class VmmException extends RuntimeException
{    
    
    private static final long serialVersionUID = 3361783299853681520L;

    public VmmException() {
        super("Native call to vmm failed!");
    }

    public VmmException(String errorMessage) {
        super(errorMessage);
    }
    
    public VmmException(String errorMessage, Throwable t) {
        super(errorMessage, t);
    }
    
}

File: vmmsharp/vmmsharp/vmmsharp/VmmException.cs
/*  
 *  C# API wrapper 'vmmsharp' for MemProcFS 'vmm.dll' and LeechCore 'leechcore.dll' APIs.
 *  
 *  Please see the example project in vmmsharp_example for additional information.
 *  
 *  Please consult the C/C++ header files vmmdll.h and leechcore.h for information about parameters and API usage.
 *  
 *  (c) Ulf Frisk, 2020-2025
 *  Author: Ulf Frisk, pcileech@frizk.net
 *  
 */

/* Contributions by imerzan (Frostchi)
 * BSD Zero Clause License
 * 
 * Copyright (c) 2024 imerzan
 * 
 * Permission to use, copy, modify, and/or distribute this software for any
 * purpose with or without fee is hereby granted.
 * 
 * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
 * REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
 * AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
 * INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
 * LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
 * OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
 * PERFORMANCE OF THIS SOFTWARE.
 */

using System;

namespace Vmmsharp
{
    /// <summary>
    /// Thrown when an exception occurs within Vmmsharp (MemProcFS).
    /// </summary>
    public class VmmException : Exception
    {
        public VmmException()
        {
        }

        public VmmException(string message)
            : base(message)
        {
        }

        public VmmException(string message, Exception inner)
            : base(message, inner)
        {
        }
    }
}
